### DISCUSSION TITLE
Broadlink RM3 Mini - Dispositivo non supportato

### DISCUSSION THREAD

**Discussion by rmoggia** [indomus/forum#582] [HUB: Home Assistant]
Buonasera. 
Ho comprato dei nuovi Broadlink R3 Mini. Ne ho già uno comprato qualche anno fa. Il primo viene visto dall'integrazione di HA.
Quando tento di aggiungere i nuovi ricevo un messaggio; `Dispositivo non supportato`. Nei log ho trovato questo errore:
`[homeassistant.components.broadlink.config_flow] Unsupported device: 0x6507. If it worked before, please open an issue at https://github.com/home-assistant/core/issues`. Ho aggiunto alla mia rete i BL con l'app Broadlink interrompendo la configurazione come suggerito nella guida indomus. Di per se i dispositivi funzionano, li ho provati con l'app giusto per togliermi il dubbio.
Qualcuno ha qualche idea, qualche suggerimento?
Grazie mille

**Comment by rmoggia** [comment]
Buongiorno. Un aggiornamento. Ho risolto il messaggio di dispositivo non supportato, occorre inserirlo all'interno dello script python che gestisce il broadlink.  Allego un immagine tratta da un forum dove ho trovato chi aveva il mio stesso problema ma con un RM4. Sono indicate le istruzioni da seguire. Occorre una sessione ssh. Ricordarsi di disabilitare la protezione ssh. Fare un backup prima di operare. 
La riga da aggiungere è:
`0x6507`: (rm3, "RM3  mini", "Broadlink")
Giusto per la precisione il mio broadlink è un R3 mini versione con cappello nero e corpo arancione.
![Immagine1](https://user-images.githubusercontent.com/58051868/127984106-8190f94f-4cae-4443-a534-4f01ae68a43e.png)

**Reply by snakuzzo** [reply]
Scusa, ma che versione di HA (o del modulo python) hai ?
Te lo chiedo perché quel modulo ha già quella riga nel file `__init__.py`

Guarda...
https://github.com/mjg59/python-broadlink/blob/822b3c326631c1902b5892a83db126291acbf0b6/broadlink/__init__.py#L90

...e l'ultimo aggiornamento di quel modulo risale a 6 mesi fa

**Reply by rmoggia** [reply]
Ciao. Non il 6539 che è marcato come rm4  ma il 6507 che è un r3mini. Comunque adesso riesco ad installarlo nel'integrazione di Broadlink, ma mi restituisce un errore di disk full. Devo approfondire. Comunque grazie

**Comment by rmoggia** [comment]
Ho risolto il problema. Beh in effetti non l'ho risolto io ma **felipediel** . La stringa da aggiungere è la seguente: `0x6507 (rmminib,"RM mini 3","Broadlink");`
Con questa modifica funziona il mio R3 mini arancione con cappello nero.
Ora ho ancora un piccolo problema. Ho perso già una volta la modifica dopo aver aggiornato il core alla successiva versione.
Vi aggiorno. Grazie

**Reply by snakuzzo** [reply]
Eh grazie...quando aggiorni Core...

Se sei in venv e c'è un aggiornamento del modulo, te lo sovrascrive (risolvibile con uno script che aggiunge quella riga)

Se sei su docker te lo sovrascrive quando scarichi l'immagine nuova (risolvibile con uno script come sopra o con un comando nel compose e o in un Dockerfile)

Ma queste sono pezze da quattro soldi.
Andate sul progetto github di questo modulo e fate una PR. Fate prima e avrete la soluzione pulita.

**Reply by rmoggia** [reply]
Immaginavo che ci fosse una soluzione definitiva, prò non sono molto esperto di informatica. Mi studierò cosa significa PR. Ci provo.
Grazie dei suggerimenti.

**Reply by snakuzzo** [reply]
Se volete controllo che non sia già stata fatta e la faccio io...ci vuole veramente un attimo eh

**Comment by rmoggia** [comment]
Saresti gentile, grazie. Non saprei proprio cosa fare.

**Reply by snakuzzo** [reply]
Fatta

https://github.com/mjg59/python-broadlink/pull/614

**Comment by rmoggia** [comment]
Innanzitutto grazie. Quindi adesso anche al prossimo aggiornamento mi ritroverò la modifica? Giusto?

**Reply by snakuzzo** [reply]
non so quando te la ritroverai.
Segui la PR e segui il progetto su github

**Reply by snakuzzo** [reply]
e per favore, rispondi sotto il messaggio come ho fatto io, non creare N thread ogni volta.
Grazie

**Reply by rmoggia** [reply]
Ok. Grazie
